tests/cmake/CMakeLists.txt

```diff
@@ -26,3 +26,8 @@ endif()
 set(LZ4_TESTS_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/..")
 add_executable(decompress-partial "${LZ4_TESTS_SOURCE_DIR}/decompress-partial.c")
 target_link_libraries(decompress-partial LZ4::lz4_shared)
+
+# Add a test that builds against the lz4::lz4 target
+set(LZ4_TESTS_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/..")
+add_executable(decompress-partial-lz4 "${LZ4_TESTS_SOURCE_DIR}/decompress-partial.c")
+target_link_libraries(decompress-partial-lz4 lz4::lz4)
```

Comment:
NB:
In lz4 CI, this will link to the shared lib even for "cmake (static lib)". Same as LZ4::lz4.
However, in vcpkg CI, the x64-linux triplet (and others) use static library linkage.